src/query/functions/temporal/base.go

```diff
@@ -223,14 +228,31 @@ func batchProcess(
 	return multiErr.FinalError()
 }
 
-func buildBlockBatch(
+func parallelProcess(
+	ctx context.Context,
 	idx int,
 	iter block.SeriesIter,
 	builder block.Builder,
 	blockMeta blockMeta,
 	processor processor,
 	mu *sync.Mutex,
 ) error {
+	var (
+		start      = time.Now()
+		decodeTime time.Duration
+	)
+	_, sp, _ := xcontext.StartSampledTraceSpan(ctx, tracepoint.TemporalDecodeParallel)
```

Comment:
So `decodeTime` will always be initially zero here, and there is not a great way to get the `instrumented` flag in this processor component. However, I think we can create the span at the end since there is a `startTime` option, and so this should achieve the same behavior you're suggesting
```
                 _, sp, _ := xcontext.StartSampledTraceSpan(ctx, ..., opentracing.StartTime(start))
		sp.FinishWithOptions(opentracing.FinishOptions{
			FinishTime: start.Add(decodeTime),
		})
```